src/telemetry/telemetryController.ts

```diff
@@ -150,8 +160,20 @@ export default class TelemetryController {
     this._segmentAnalytics?.flush();
   }
 
-  public needTelemetry() {
-    return vscode.workspace.getConfiguration('mdb').get('sendTelemetry');
+  private needTelemetry(): boolean {
+    if (this._needTelemetry !== true) {
```

Comment:
Isn't `!this._needTelemetry` more immediate?

What about to disambiguate a bit, we could call this `shouldSendTelemetry`, `_needTelemetry` could be something like `telemetryFeatureEnabled` and `confNeedTelemetry` could be `telemetryEnabledByUser`?

Maybe i'm giving horrible suggestions here.. is just to make sure is understandable that there are 3 different things and we call them in a different way:
- a feature flag to disable the telemetry during tests
- a configuration for the user
- something to really decide wether the telemetry should be sent or not.

Reply:
Love your suggestions! ❤️ Also added comments to make it absolutely clear what is what.